synapse/handlers/presence.py

```diff
@@ -1764,14 +1764,14 @@ async def _filter_all_presence_updates_for_user(
         Returns:
             A list of presence states for the given user to receive.
         """
+        updated_users = None
         if from_key:
             # Only return updates since the last sync
             updated_users = self.store.presence_stream_cache.get_all_entities_changed(
                 from_key
             )
-            if not updated_users:
-                updated_users = []
```

Comment:
`_filter_app_presence_updates_for_user` is only called if `additional_users_interested_in == PresenceRouter.ALL_USERS`, so it seems reasonable here to treated this the same as not having a `from_key` (and returning info on all users).